tests/PHPStan/Rules/Functions/data/function-call-statement-no-side-effects-8.0.php

```diff
@@ -19,6 +19,11 @@ public function noEffect(string $url, $resourceOrNull)
 		var_export([], return: true);
 		print_r([], return: true);
 		highlight_string($url, return: true);
+		array_filter([], callback: 'is_string');
```

Comment:
We should test cases in which the callback is impure, e.g. because of calls to impure functions or access to global variables or similar